pkg/api/v1/types.go

```diff
@@ -41,6 +41,9 @@ type Properties struct {
 	// RouterProfiles (in,optional/out): Configuration for OpenShift router(s).
 	RouterProfiles []RouterProfile `json:"routerProfiles,omitempty"`
 
+	// MasterPoolProfile (in): Configuration for OpenShift master VMs.
+	MasterPoolProfile MasterPoolProfile `json:"masterPoolProfile,omitempty"`
```

Comment:
Note for reviewers: this name was chosen because we have a `MasterProfile` on the internal spec already and I did not want to cause confusion.

Reply:
Same applies to other fields as well so this can be `MasterProfile`.